### ISSUE TITLE
continue on error without crash if script has syntax error

### ISSUE THREAD

**Issue by kenyeung128** [bwilcox-1234/ChatScript#156]
Hi, is there anyway to prevent the ChatScript system from crashing if there is a syntax error for example? as we are updating the script dynamically and sometimes if there is an error, the system will just crash, is there any way to keep it running? thanks.

**Comment by bwilcox-1234** [comment]
I dont generally experience syntax errors as crashes. Is this Windows/Mac/Linux, and can you provide a sample file which causes a crash?

**Comment by iou2much** [comment]
Hi, @bwilcox-1234 . Here's the use case.  We have multiple bots, whose topic files are  UCG. 
One of them including a line like: 


t: test ( test ) ^repeat() <p>here's your info </p><p><a href="https://google.com/" target="_blank">查閱個人資料（TEST-TEXT)</a>&nbsp;&nbsp;<br></p> 

the bracket around "TEST-TEXT" cause problem.
The left one is Full-width character and the right one is half-width character

this would cause error when execute :build command.

> ERROR SUMMARY:
>   line 22 of abc.top:
> OUTPUT-3 Unbalanced )
> 
> 1 errors - press Enter to quit. Then fix and try again.

After that it alway prompt "No such bot."
And this effects other bots with correct syntax, cuz the engine couldn't find any bots.

**Comment by bwilcox-1234** [comment]
no.  a build is everything you compile at once and you cannot composite multiple bots EXCEPT that you could have a bot in layer 0 and another in layer 1.  But... Why is your paren  different? do you need funny paren?

**Comment by bwilcox-1234** [comment]
since these are not real parens (they dont intent chatscript functionality), you could backslash them.

**Comment by iou2much** [comment]
backslash the special character is a good idea. but is it possible to skip the bot if it's corrupt? Since our script are user-generated, there might always be something we can't control in the script.  And the bot our clients build are not related to each other, so they shouldn't be in different layers, I think.

**Comment by bwilcox-1234** [comment]
CS has no way of knowing you are building 1 bot or 100 when building a layer. It's just a bunch of files in various orders.

**Comment by bwilcox-1234** [comment]
You could build tools to figure out what bot fails and build the others as a different build presuambly
